py/server/deephaven_internal/stream.py

```diff
@@ -0,0 +1,89 @@
+import io
+
+
+class TeeStream(io.TextIOBase):
+    """
+    TextIOBase subclass that splits output between a delegate instance and a set of lambdas. Can delegate some calls to
+    the provided stream but actually write only to the lambdas.
+
+    Useful to adapt any existing file-like object (such as sys.stdout or sys.stderr), but actually let some other
+    processing take place before writing.
+    """
+
+    encoding = 'UTF-8'
+    closed = False
+
+    @staticmethod
+    def split(py_stream, java_stream):
+        return TeeStream(
+            py_stream,
+            True,
+            lambda t: java_stream.write(bytes(t, py_stream.encoding)),
+            lambda: java_stream.flush(),
+            lambda: java_stream.close()
+        )
+
+    @staticmethod
+    def redirect(py_stream, java_stream):
+        if hasattr(py_stream, "encoding"):
+            encoding = py_stream.encoding
+        else:
+            encoding = 'UTF-8'
+        return TeeStream(
+            py_stream,
+            True,
```

Comment:
Should this be `False`? - I don't see `should_write_to_orig_stream` being used differently anywhere.

Reply:
It is False for `redirect()`, and True for `split`, and TeeStream has a field for it, so that write/flush/close can either pass through to the original stream, or ignore the original stream (and only delegate methods that query for characteristics, like isatty(), fileno(), and the encoding/closed properties).

There may well be a better way to handle this, but I think it is readable enough? Maybe use named params here instead of positional ones?

Reply:
Am I missing something? This is `redirect`, and the value is `True` here; and in `split`.

Reply:
Whoops - it should definitely be False for redirect. Probably got switch in my "am I going crazy" debugging around jpy.